### PR TITLE
[10.0][ADD] Module maintenance_settings

### PR DESCRIPTION
**Pull request by espo-tony**
This module doesn't introduce new functionalities; it rather provides maintenance-related generic tools to be used in further specific-purpose modules, including security groups and configuration settings section.

### CODE REVIEW

**Comment by espo-tony** [comment]
@dreispt 
Can I ask you to review this PR? The goal is not to provide new functionalities but to introduce some generic tool (like the maintenance configuration settings and domain-specific security groups) to be used in further modules.
Something like we discussed in #4 
Thank you in advance.

**Comment by pedrobaeza** [comment]
As it's a base module without functionality itself, then you should prefix it with `base_`, and avoid plurals (check https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules). Then my suggestion is `base_maintenance_config`.

**Comment by espo-tony** [comment]
@pedrobaeza 
Thank you for the suggestion. You're right, indeed. I renamed the module

File: base_maintenance_config/README.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
```
Comment (astirpe) [inline base_maintenance_config/README.rst]: According with the updated template, this line should be:
```
   :target: http://www.gnu.org/licenses/agpl
```

File: base_maintenance_config/README.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+=======================
+Base Maintenance Config
+=======================
+
+This module doesn't introduce new functionalities; it rather provides maintenance-related generic tools to be used in further specific-purpose modules.
+Such generic tools include:
+#. A general settings section for the maintenance functionalities.
+#. A security group category 'Maintenance' with two levels: 'User' and 'Manager'.
+
+Configuration
```
Comment (astirpe) [inline base_maintenance_config/README.rst]: Remove empty sections

File: base_maintenance_config/README.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+help us smash it by providing detailed and welcomed feedback.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Antonio Esposito <a.esposito@onestein.nl>
+
```
Comment (astirpe) [inline base_maintenance_config/README.rst:42]: Add this line:

```
Do not contact contributors directly about support or help with technical issues.
```

File: base_maintenance_config/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
```
Comment (astirpe) [inline base_maintenance_config/__init__.py]: Remove copyright in `__init__.py` files

File: base_maintenance_config/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
```
Comment (astirpe) [inline base_maintenance_config/models/__init__.py]: Remove copyright in `__init__.py `files

File: base_maintenance_config/menuitems.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+
+        <menuitem id="menu_stock_general_settings" name="Settings" parent="maintenance.menu_maintenance_configuration"
+            sequence="100" action="action_maintenance_config_settings" groups="maintenance.group_equipment_manager"/>
+
+        <record model="ir.ui.menu" id="maintenance.menu_maintenance_configuration">
```
Comment (astirpe) [inline base_maintenance_config/menuitems.xml]: The `id` definition must precede the `model`, as stated in the OCA guidelines

File: base_maintenance_config/views/maintenance_config_settings.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (astirpe) [inline base_maintenance_config/views/maintenance_config_settings.xml]: `<data>` could be removed

File: base_maintenance_config/menuitems.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (astirpe) [inline base_maintenance_config/menuitems.xml]: `<data>` could be removed

**Review by astirpe — CHANGES_REQUESTED** [review]

File: base_maintenance_config/README.rst [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+#. In the tab 'Access Rights', at the 'Application' paragraph you'll now be able to select a group, including 'User' and 'Manager' for the 'Maintenance' category
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/92/10.0
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
```
Comment (astirpe) [inline base_maintenance_config/README.rst:30]: Replace `account-financial-tools` with `maintenance` in this link

File: base_maintenance_config/README.rst [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+This module doesn't provide new functionalities.
+Still you can now set a new security group on users. To do so, you need to:
+#. Go on 'Settings' -> 'Users' -> 'Users'
+#. Create a new user or select an already existing one
+#. In the tab 'Access Rights', at the 'Application' paragraph you'll now be able to select a group, including 'User' and 'Manager' for the 'Maintenance' category
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/92/10.0
```
Comment (astirpe) [inline base_maintenance_config/README.rst]: Replace id 92 with the one for maintenance

**Review by astirpe — CHANGES_REQUESTED** [review]

**Review by astirpe — APPROVED** [review]

File: base_maintenance_config/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Base Maintenance Config',
+    'summary': 'Provides general settings for the Maintenance App',
+    'author': 'Onestein, Odoo Community Association (OCA)',
+    'website': 'http://www.onestein.eu',
```
Comment (MiquelRForgeFlow) [inline base_maintenance_config/__manifest__.py]: `https://github.com/OCA/maintenance`

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1:

**Review by dreispt — APPROVED** [review]

**Pull request merged**